src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,181 @@
+import { RequestSender } from '@bigcommerce/request-sender';
+
+import { PaymentStrategy } from '../';
+import {
+    Payment,
+    PaymentActionCreator,
+    PaymentInitializeOptions,
+    PaymentMethodActionCreator,
+    PaymentRequestOptions,
+    PaymentStrategyActionCreator
+} from '../..';
+import { CheckoutActionCreator, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { NotInitializedError } from '../../../common/error/errors';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType,
+    NotInitializedErrorType,
+} from '../../../common/error/errors';
+import { toFormUrlEncoded } from '../../../common/http-request';
+import { bindDecorator as bind } from '../../../common/utility';
+import {
+    OrderActionCreator, OrderRequestBody } from '../../../order';
+
+import { GooglePayPaymentInitializeOptions, GooglePayPaymentProcessor } from './';
+import {
+    GooglePaymentData,
+    GooglePayAddress,
+    GooglePayInitializer,
+    PaymentMethodData,
+    TokenizePayload,
+} from './googlepay';
+
+export default class GooglePayPaymentStrategy extends PaymentStrategy {
+    private _googlePayOptions!: GooglePayPaymentInitializeOptions;
+    private _methodId!: string;
+    private _walletButton?: HTMLElement;
+
+    constructor(
+        store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _googlePayInitializer: GooglePayInitializer,
```

Comment:
Why do you need to inject `GooglePayInitializer` here? You don't need to because you already have `GooglePayPaymentProcessor`, which acts as a proxy for the former.